### PR TITLE
move the wal repair logic in db.Open

### PR DESCRIPTION
**Pull request by krasi-georgiev**
This is to allow opening a wal in a read only mode without triggering a
repair.

This is as part of working on a PR to add opening a db in a read only mode.

https://github.com/prometheus/tsdb/pull/588

cc @codesome 	trying to brake the huge PR into few smaller ones.

### CODE REVIEW

File: wal/wal.go [hunk 1]
```diff
@@ -289,6 +294,7 @@ func (w *WAL) Repair(origErr error) error {
 	if !ok {
 		return errors.Wrap(origErr, "cannot handle error")
 	}
+	w.walCorruptionsTotal.Inc()
```
Comment (codesome) [inline wal/wal.go]: Is it safe to assume that `Repair` is always called after corruption? (Should not regress in future as this is a separate package)

Comment (krasi-georgiev) [reply wal/wal.go]: sorry I didn't understand what you mean. 
I think the metric is to notify for corruption, regardless if it has been repaired or not.

**Comment by krasi-georgiev** [comment]
> Maybe you can also mention the accidental bugfix in the CHANGELOG.

I think there was no actual bug after all as wal.Repair would delete all wal files after the corrupted one so the reader wouldn't be able to read these anyway.

I think I even checked that one of the test covers such case and it wasn't failing.

File: wal/wal.go (hunk 1 above)
Comment (codesome) [reply wal/wal.go]: Yes, so it should not happen in the future that we encounter a corruption but don't call `Repair` due to some reasons. In that case, corruption is not counted. Would it be better to increment the counter right when we encounter corruption?

Comment (krasi-georgiev) [reply wal/wal.go]: yeah I see you point. I moved it.

**Pull request merged**